guide/oauth2/README.md

```diff
@@ -0,0 +1,200 @@
+# OAuth2 
+
+OAuth2 enables application developers to build applications that utilize authentication and data from the Discord API.
```

Comment:
The intro section is a bit bland. I think adding another sentence saying something along the lines of "It can be used to create things such as web dashboard to display user info, statistics, and allow extended functionality to your bot" would help interest the user.